docs/references/calfits_memo.tex

```diff
@@ -0,0 +1,183 @@
+\geometry{letterpaper}                   		% ... or a4paper or a5paper or ... 
+%\geometry{landscape}                		% Activate for for rotated page geometry
+%\usepackage[parfill]{parskip}    		% Activate to begin paragraphs with an empty line rather than an indent
+\usepackage{graphicx}			
+							
+\usepackage{amssymb}
+
+\usepackage{hyperref} 
+\hypersetup{
+    colorlinks = true
+}
+
+\title{Memo : pyuvdata.uvcal}
+\author{Zaki Ali, Bryna Hazelton, Adam Beardsley, Paul La Plante and the pyuvdata team}
+\date{July 3, 2017}							% Activate to display a given date or no date
+
+\begin{document}
+\maketitle
+\section{Introduction}
+This memo introduces the new \textit{calfits} file format for storing
+calibration solutions using
+pyuvdata\footnote{\url{https://github.com/HERA-Team/pyuvdata}}, a python package that
+provides an interface to interferometric data. We are defining a file format
+with the code interface. Here, we describe the required and optional parameters
+of the \textit{calfits} format and interface to reading and writing, in addition
+to the structure of the underlying fits file. For examples please see the
+pyuvdata tutorial on ReadTheDocs: \url{http://pyuvdata.readthedocs.io/en/latest/tutorial.html}.
+
+\section{Basics}
+
+The \textit{calfits} format is a specification for the storage of radio
+interferometer calibration information in a fits file.  The contents of the file
+are defined via explicit mapping to the UVCal object in the pyuvdata package.
+The UVCal object (really, it's a class) is a subclass of the of the UVBase class
+with a set of \textbf{uvparameters} that defines the UVCal object. A
+\textbf{uvparameter} is a pyuvdata object that has a name, form, description,
+value, required flag, expected type, acceptable values, and tolerances
+associated with it.  \textbf{Uvparameters} are accessed as attributes to the
+UVBase class (and its subclasses, like UVCal).
+
+\section{Parameters}
+In order to conform to the \textit{calfits} file format there are a number of
+required parameters that need to be set as attributes of the UVCal class. Below
+is a list of the required parameters and their descriptions.
+
+\begin{itemize}
+\item{\textbf{Nants\_data}: Number of antennas with data present. May be smaller
+    than the number of antennas in the array}
+\item{\textbf{Nants\_telescope}: Number of antennas in the array. May be larger
+    than the number of antennas with data}
+\item{\textbf{Nfreqs}: Number of frequency channels. Note that for
+    cal\_type=delay, Nfreq=1 as a placeholder, so that arrays for both
+    cal\_types have the same rank.}
+\item{\textbf{Njones}: Number of polarizations calibration parameters (Number of
+    jones matrix elements.).}
+\item{\textbf{Nspws}: Number of spectral windows (ie non-contiguous spectral
+    chunks). More than one spectral window is not currently supported.}
+\item{\textbf{Ntimes}: Number of times}
```

Comment:
That's not quite right because sometimes a single calibration solution is calculated over many integrations. In that case, Ntimes would be 1 but would not represent the number of integrations. I tried to clarify.